draft/src/composite_drafter.rs

```diff
@@ -0,0 +1,109 @@
+use crate::api::*;
+use std::collections::HashMap;
+pub struct CompositeDrafter {
+
+}
+
+// Draft brains will make determinations on how to draft based
+// on various metrics provided to them within card data.
+
+/* A rudimentary brain that counts mana symbols in mana costs and attempts to find cards
+ with average color similar to that of the existing cards in our deck. */
+pub struct DraftColorBrain {
+}
+
+impl DraftColorBrain {
+    // Function that builds a hash map of colors and their weights; higher weights are closer to our
+    // current colors.
+    // TODO: In the python conversion, this state should be kept and updated incrementally, rather
+    // than recalculated wholesale every pick.
+    fn calc_preferences(&self, cards_owned: &[Card]) -> HashMap<BasicMana, i32> {
+        let mut mana_prefs: HashMap<BasicMana, i32> = HashMap::new();
+        for i in 0..cards_owned.len() {
+            for symbol in 0..cards_owned[i].mana_cost.cost.len() {
+                // TODO: Find a good way to return the primitive mana types of complex mana types
+                // and increment our hashmap values accordingly.
+                match &cards_owned[i].mana_cost.cost[symbol] {
+                    Mana::XCost(string) => (),
+                    Mana::Phyrexian(basic) => {mana_prefs.entry(basic.clone()).and_modify(|m| { *m += 1 }).or_insert(1);},
+                    Mana::Split(basic, basic2) => {
+                        mana_prefs.entry(basic.clone()).and_modify(|m| { *m += 1 }).or_insert(1);
+                        mana_prefs.entry(basic2.clone()).and_modify(|m| { *m += 1 }).or_insert(1);
+                    },
+                    Mana::Standard(basic) => {mana_prefs.entry(basic.clone()).and_modify(|m| { *m += 2 }).or_insert(2);},
+                    Mana::NumGeneric(amt) => (),
+                    Mana::SplitGeneric(basic) => {mana_prefs.entry(basic.clone()).and_modify(|m| { *m += 1 }).or_insert(1);},
+                }
+            }
+        }
+        mana_prefs
+    }
+
+    // In truth I think this strategy is terrible and will move to clustering for python,
```

Comment:
Can you also summarize what the current strategy is in this comment? From what I can tell, if my card is RBB, and my prefs have red: 10 and black: 7, then I end up with 24, is that right?